smpp/server.go

```diff
@@ -127,6 +125,21 @@ func NewUnstartedServer(user, password string, listener net.Listener) Server {
 	return s
 }
 
+// NewTLDUnstartedServer creates a new SSL Server with default settings, and
+// does not start it. Callers are supposed to call Start and Close later.
+func NewTLSUnstartedServer(user, password string, listener net.Listener, tlsConf *tls.Config) Server {
+	s := &server{
+		User:   user,
+		Passwd: password,
+		TLS:    tlsConf,
+		m:      map[pdu.ID]RequestHandlerFunc{},
+		a:      map[pdu.ID]AuthRequestHandlerFunc{},
+		s:      map[string]Session{},
+		l:      listener,
+	}
+	return s
+}
+
```

Comment:
Just realised the whole server bit of our go-smpp was copy-pasted from the fiorix test. I assumed that if server had a tls field it's probably implemented and should just be populated, apparently it's not :(. Not sure what their purpose was then. I'll look into it now.